tools/stylelint/no-concrete-rules.ts

```diff
@@ -1,29 +1,40 @@
-const stylelint = require('stylelint');
-const path = require('path');
+import {createPlugin, utils} from 'stylelint';
+import {basename} from 'path';
+
 const ruleName = 'material/no-concrete-rules';
-const messages = stylelint.utils.ruleMessages(ruleName, {
+const messages = utils.ruleMessages(ruleName, {
   expected: pattern => `CSS rules must be placed inside a mixin for files matching '${pattern}'.`
 });
 
+/** Config options for the rule. */
+interface RuleOptions {
+  filePattern: string;
+}
+
 /**
  * Stylelint plugin that will log a warning for all top-level CSS rules.
  * Can be used in theme files to ensure that everything is inside a mixin.
  */
-const plugin = stylelint.createPlugin(ruleName, (isEnabled, options) => {
+const plugin = createPlugin(ruleName, (isEnabled: boolean, _options) => {
   return (root, result) => {
-    if (!isEnabled) return;
+    if (!isEnabled) {
+      return;
+    }
 
+    const options = _options as RuleOptions;
     const filePattern = new RegExp(options.filePattern);
-    const fileName = path.basename(root.source.input.file);
+    const fileName = basename(root.source!.input.file!);
```

Comment:
In what cases can this be null, do you know?

Reply:
No idea since we've never had a null check around it and it hasn't been throwing. For some reason the typings have it nullable.

Reply:
Just looked at their source. It looks like it is nullable because in-memory files `Node´'s could be created.